admin/spec/service/check-window.service.spec.js

```diff
@@ -189,5 +189,55 @@ describe('check-window.service', () => {
       expect(result.toString).toBe(expected.toString)
       expect(result).toBeTruthy()
     })
+    describe('#isLoginAllowed', () => {
+      it('should allow pupil to login if within range of active check window', async (done) => {
+        spyOn(checkWindowDataService, 'sqlFindOneCurrent').and.returnValue({
+          checkStartDate: moment.utc().subtract(1, 'days'),
+          checkEndDate: moment.utc().add(1, 'days')
+        })
+        try {
+          await service.isLoginAllowed()
+        } catch (error) {
+          fail('not expected to throw')
+        }
+        done()
+      })
+      it('should disallow pupil to login if there is now active window', async (done) => {
+        spyOn(checkWindowDataService, 'sqlFindOneCurrent').and.returnValue(Promise.resolve({}))
+        try {
+          await service.isLoginAllowed()
+          fail('not expected to throw')
+        } catch (error) {
+          expect(error.message).toBe('There is no active check window')
+        }
+        done()
```

Comment:
You know, as we are not using callbacks, there is no need for `done` to be here.  `await` removed the need for it.

Reply:
@jon-shipley i thought I had them removed. Guess I didn't. Updated